arrow-core-data/src/main/kotlin/arrow/core/ListK.kt

```diff
@@ -167,14 +167,41 @@ data class ListK<out A>(private val list: List<A>) : ListKOf<A>, List<A> by list
       }
     }
 
+  @Deprecated("Deprecated, use filterNullMap(f: (A) -> B?) instead", ReplaceWith("filterNullMap(f: (A) -> B?)"))
   fun <B> filterMap(f: (A) -> Option<B>): ListK<B> =
     flatMap { a -> f(a).fold({ empty<B>() }, { just(it) }) }
 
+  /**
+   * Returns a [ListK] containing the transformed values from the original
+   * [ListK] as long as they are non-null.
+   *
+   * Example:
+   * ```kotlin:ank:playground
+   * import arrow.core.*
+   *
+   * //sampleStart
+   * val evenStrings = listOf(1, 2).k().filterNullMap {
+   *   when (it % 2 == 0) {
+   *     true -> it.toString()
+   *     else -> null
+   *   }
+   * } // Result: listOf("2").k()
+   * //sampleEnd
+   *
+   * fun main() {
+   *   println("evenStrings = $evenStrings")
+   * }
+   * ```
+   */
+  fun <B> filterNullMap(f: (A) -> B?): ListK<B> =
```

Comment:
Should we do? We're also trying to algining more to the names familiar in Kotlin from the std. IIRC there is 2 `Iterable` extensions for the signature in the std. `filterMap` and `mapNotNull`?

```
@JvmName("mapNotNull")
fun <B> filterMap(f: (A) -> B?): ListK<B>
```

Reply:
I think as long as the stdlib ones don't have any obvious drawbacks we shouldn't have duplicates names and signatures.

Reply:
In that case I think we should name it [`mapNotNull` as Kotlin's Std](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/map-not-null.html)? 

We can re-introduce `filterMap` if we want later on.